### PR TITLE
Provide UUIDField option for pk field and a natural key feature for serialization

### PR DESCRIPTION
**Pull request by mathiasag7**
#427 Feature

Only things haven't written a test case for it. I would have been happy to do it but still learning how to do so.

### CODE REVIEW

**Comment by Dresdn** [comment]
Thanks for the PR @mathiasag7. Could you revert back changes that don't specifically apply to the feature? For example newlines, quotes, language changes, etc.? I'll mark them as comments.

File: pyproject.toml
Comment (Dresdn) [inline pyproject.toml:1]: The versioning is handled by the release, so no need to update anything here.

File: test_project/settings.py [hunk 1]
```diff
@@ -65,11 +68,13 @@
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': ':memory:',
-    },
+        'NAME': BASE_DIR / 'db.sqlite3',
```
Comment (Dresdn) [inline test_project/settings.py]: Let's revert this too to not create clutter just to run tests.

File: test_project/settings.py [hunk 2]
```diff
@@ -65,11 +68,13 @@
...
+    }
 }
 
+
 DEFAULT_AUTO_FIELD = 'django.db.models.AutoField'
+PRIMARY_KEY_FIELD = 'django.db.models.AutoField'
```
Comment (Dresdn) [inline test_project/settings.py]: The setting name being `PRIMARY_KEY_FIELD` that's used just for the project can be more descriptive. Maybe this gets updated to something like `EAV2_PRIMARY_KEY_FIELD` or `DJANGO_EAV2_PK_TYPE`

File: README.md
Comment (Dresdn) [inline README.md:1]: I was going to comment that this level of detail should go in the docs, but realized the docs build has been failing. If you are able to move this to the docs, great, if not, don't worry about it. This is more of a note for me then.

File: eav/logic/managers.py
Comment (Dresdn) [inline eav/logic/managers.py:1]: @mathiasag7 - What's the reasoning behind the `get_by_natural_key()` methods?

`Attribute.objects.get_by_natural_key(name='age', slug='age')` is the same as `Attribute.objects.get(name='age', slug='age')`

File: eav/models.py [hunk 3]
```diff
@@ -288,6 +317,12 @@ class Meta:
         verbose_name=_('Created'),
     )
 
+    def natural_key(self):
```
Comment (Dresdn) [inline eav/models.py]: Same with these. Is there value there for having these methods vs doing `(attr.name, attr.slug)`?

**Review by Dresdn — CHANGES_REQUESTED** [review]
Apologies for the delay on the feedback. I'm recovering from being ill so am catching up on things.

File: eav/logic/managers.py
Comment (mathiasag7) [reply eav/logic/managers.py:1]: ok I understand, and I wish you get well very soon.
About that one , get_by_natural_key is a method implemented in the manager and used for deserialisation of object with django. 
[Deserialization of natural keys](https://docs.djangoproject.com/en/4.2/topics/serialization/#deserialization-of-natural-keys).

File: eav/models.py (hunk 3 above)
Comment (mathiasag7) [reply eav/models.py]: [Django Natural keys](https://docs.djangoproject.com/en/4.2/topics/serialization/#natural-keys).

**Comment by mathiasag7** [comment]
I actually thought it might be useful, but the idea came from a project we were working on where we needed to copy data from one database to another from time to time, and to do this we chose django's serialization / deserialization process. To do this, we chose to serialize and deserialize objects. To this we coupled the use of django's feature natural key, which allows unique values to be used instead of pk during serialization. However, if this feature isn't crucial and necessary, we can remove it.

**Comment by Dresdn** [comment]
Thanks for the explanation. Let's keep it but call it out as a new feature. Thoughts?

Edit: Doh! I see you already did that. I'm going to update some docstrings and then get this merged.

**Review by Dresdn — APPROVED** [review]

**Comment by mathiasag7** [comment]
Hello, I just wanted to inform you that this version of django eav is not yet available to be installed directly with pip in a project.

**Pull request merged**